app/views/posts/_expanded.html.erb

```diff
@@ -293,6 +293,33 @@
           </div>
         <% end %>
 
+        <% if is_question && post.children.length >= SiteSetting["TableOfContentsThreshold"] %>
+          <div class="toc" id="toc-example-toggle">
+            <button class="toc--header" data-toggle="#toc-example-toggle" data-toggle-property="class" data-toggle-value="is-active">Table of Contents</button>
+            <% post.children.each do |answer| %>
+              <a href="#" class="toc--entry">
+                <div class="toc--badge">
+                  <span class="badge is-tag is-green">+<%= answer.upvote_count %></span>
+                </div>
+                <div class="toc--badge">
+                  <span class="badge is-tag is-red">&minus;<%= answer.downvote_count %></span>
+                </div>
+                <div class="toc--full">
+                  <% sanitised = raw(sanitize(answer.body, scrubber: scrubber)) %>
+                  <% max_length = SiteSetting["TableOfContentsExcerptLength"] %>
+                  <p>
+                    <%=raw sanitised[ 0 ... max_length ] %>
+                    <% if sanitised.length > max_length %>
+                      ...
```

Comment:
[**nitpick (non-blocking):**](https://conventionalcomments.org/) It may be better to use the HTML `&hellip;` entity for semantical/accessibility reasons. OTOH we aren't doing it elsewhere, so this is probably fine.